queries.txt

```diff
@@ -0,0 +1,211 @@
+
+--1) List all the columns of the Salespeople table.
+SELECT * FROM salespeople;
+
+ snum |  sname  |   city    | comm 
+------+---------+-----------+------
+ 1001 | Peel    | London    |   12
+ 1002 | Serres  | SanJose   |   13
+ 1004 | Motika  | London    |   11
+ 1007 | Rifkin  | Barcelona |   15
+ 1003 | AxelRod | NeyYork   |   10
+ 1008 | Fran    | London    |   25
+(6 rows)
+
+--2) List all customers with a rating of 100.
+SELECT CNAME from customers
+WHERE rating = 100;
+
+ cname  
+---------
+ Hoffman
+ Clemens
+ Pereira
+(3 rows)
+
+--3) Find the largest order taken by each salesperson on each date.
+SELECT snum,odate,MAX(amt)
+FROM orders
+GROUP BY snum,odate
+ORDER BY snum;
+
+ snum |   odate    |   max   
+------+------------+---------
+ 1001 | 1990-05-10 | 4723.00
+ 1001 | 1990-06-10 | 9891.88
+ 1001 | 1990-03-10 |  767.19
+ 1002 | 1990-06-10 | 1309.95
+ 1002 | 1990-03-10 | 5160.45
+ 1002 | 1990-04-10 |   75.75
+ 1003 | 1990-04-10 | 1713.23
+ 1004 | 1990-03-10 | 1900.10
+ 1007 | 1990-03-10 | 1098.16
+(9 rows)
+
+--4) Arrange the Order table by descending customer number.
+SELECT * FROM orders
+ORDER BY cnum DESC;
+
+ onum |   amt   |   odate    | cnum | snum 
+------+---------+------------+------+------
+ 3006 | 1098.16 | 1990-03-10 | 2008 | 1007
+ 3001 |   18.69 | 1990-03-10 | 2008 | 1007
+ 3002 | 1900.10 | 1990-03-10 | 2007 | 1004
+ 3011 | 9891.88 | 1990-06-10 | 2006 | 1001
+ 3008 | 4723.00 | 1990-05-10 | 2006 | 1001
+ 3007 |   75.75 | 1990-04-10 | 2004 | 1002
+ 3010 | 1309.95 | 1990-06-10 | 2004 | 1002
+ 3005 | 5160.45 | 1990-03-10 | 2003 | 1002
+ 3009 | 1713.23 | 1990-04-10 | 2002 | 1003
+ 3003 |  767.19 | 1990-03-10 | 2001 | 1001
+(10 rows)
+
+--5) Find which salespeople currently have orders in the order table.
+SELECT DISTINCT(snum),sname 
+FROM orders NATURAL JOIN salespeople;
+
+ snum |  sname  
+------+---------
+ 1004 | Motika
+ 1003 | AxelRod
+ 1007 | Rifkin
+ 1001 | Peel
+ 1002 | Serres
+(5 rows)
+
+--6) List names of all customers matched with the salespeople serving them.
+SELECT cnum,cname,sname
+FROM salespeople JOIN customers
+ON salespeople.snum = customers.snum;
+
+cnum |  cname   |  sname  
+------+----------+---------
+ 2001 | Hoffman  | Peel
+ 2002 | Giovanni | AxelRod
+ 2003 | Liu      | Serres
+ 2004 | Grass    | Serres
+ 2006 | Clemens  | Peel
+ 2008 | Cisneros | Rifkin
+ 2007 | Pereira  | Motika
+(7 rows)
+
+--7) Find the names and numbers of all salespeople who have more than one customer.
+SELECT snum,sname
+FROM salespeople
+WHERE snum IN (SELECT customers.snum
+FROM customers JOIN salespeople
+ON customers.snum = salespeople.snum
+GROUP BY customers.snum
+HAVING COUNT(cnum) > 1);
+
+ snum | sname  
+------+--------
+ 1001 | Peel
+ 1002 | Serres
+(2 rows)
+
+--8) Count the orders of each of the salespeople and output the results in descending order.
+SELECT salespeople.snum,SUM(AMT)
+FROM  salespeople LEFT OUTER JOIN orders 
+ON salespeople.snum = orders.snum
+GROUP BY salespeople.snum;
+
```

Comment:
The salesperson's name should be displayed here.

Reply:
> The salesperson's name should be displayed here.
Thankyou Sir for your feedback,
I will update the query.
Regards.